src/foundry/client-esm/nue/tour.d.mts

```diff
@@ -1,9 +1,276 @@
+
+  set id(value: string | undefined);
+
+  /**
+   * The human-readable title for the tour.
+   */
+  get title(): string;
+
+  /**
+   * The human-readable description of the tour.
+   */
+  get description(): string | undefined;
+
+  /**
+   * The package namespace for the tour.
+   */
+  get namespace(): string | undefined;
+
+  set namespace(value: string | undefined);
+
+  /**
+   * The key the Tour is stored under in game.tours, of the form `${namespace}.${id}`
+   */
+  get key(): string;
+
+  /**
+   * The configuration of tour steps
+   */
+  get steps(): Tour.Step[];
+
+  /**
+   * Return the current Step, or null if the tour has not yet started.
+   */
+  get currentStep(): Tour.Step | null;
+
+  /**
+   * The index of the current step; -1 if the tour has not yet started, or null if the tour is finished.
+   */
+  get stepIndex(): number | null;
+
+  /**
+   * Returns True if there is a next TourStep
+   */
+  get hasNext(): boolean;
+
+  /**
+   * Returns True if there is a previous TourStep
+   */
+  get hasPrevious(): boolean;
+
+  /**
+   * Return whether this Tour is currently eligible to be started?
+   * This is useful for tours which can only be used in certain circumnstances, like if the canvas is active.
+   */
+  get canStart(): boolean;
+
+  /**
+   * The current status of the Tour
+   */
+  get status(): Tour.STATUS;
+
+  /**
+   * Advance the tour to a completed state.
+   */
+  complete(): Promise<void>;
+
+  /**
+   * Exit the tour at the current step.
+   */
+  exit(): void;
+
+  /**
+   * Reset the Tour to an un-started state.
+   */
+  reset(): Promise<void>;
+
+  /**
+   * Start the Tour at its current step, or at the beginning if the tour has not yet started.
+   */
+  start(): Promise<void>;
+
+  /**
+   * Progress the Tour to the next step.
+   */
+  next(): Promise<void>;
+
+  /**
+   * Reward the Tour to the previous step.
+   */
+  previous(): Promise<void>;
+
+  /**
+   * Progresses to a given Step
+   * @param stepIndex - The step to progress to
+   */
+  progress(stepIndex: number): Promise<void>;
+
+  /**
+   * Query the DOM for the target element using the provided selector
+   * @param selector - A CSS selector
+   * @returns The target element, or null if not found
+   */
+  protected _getTargetElement(selector: string): Element | null;
+
+  /**
+   * Creates and returns a Tour by loading a JSON file
+   * @param filepath The path to the JSON file
+   */
+  static fromJSON(filepath: string): Promise<Tour>;
+
+  /**
+   * Set-up operations performed before a step is shown.
+   * @virtual
+   */
+  protected _preStep(): Promise<void>;
+
+  /**
+   * Clean-up operations performed after a step is completed.
+   * @virtual
+   */
+  protected _postStep(): Promise<void>;
+
+  /**
+   * Renders the current Step of the Tour
+   */
+  protected _renderStep(): Promise<void>;
+
+  /**
+   * Reloads the Tour's current step from the saved progress
+   * @internal
+   */
+  _reloadProgress(): void;
+}
```

Comment:
Nit; It basically doesn't matter but when a class has a truly private property there's two possible noticeable changes:
1. The class becomes nominal. Normally TypeScript is okay with this:
```ts
declare class Foo {
    prop: number;
}

const foo: Foo = { prop: 123 };
```

Because TypeScript is structural. However a truly private property means this isn't allowed anymore.
2. Technically it's noticeable through variance changes but that's only relevant to generic classes.

The convention has become adding a `#tour: true` truly private property at the end (or start, we're not entirely consistent) of the class.